### PR TITLE
add backend to specify analysis metadata categories

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

**Comment by antgonza** [comment]
@wasade, this is the backend to select which metadata categories should an analysis use.

File: qiita_db/analysis.py [hunk 1]
```diff
@@ -804,6 +836,8 @@ def build_files(self, merge_duplicated_sample_ids):
             If the duplicated sample ids in the selected studies should be
             merged or prepended with the artifact ids. If false prepends
             the artifact id
+        categories : set of str
```
Comment (wasade) [inline qiita_db/analysis.py:839]:
```suggestion
        categories : set of str, optional
```

**Review by wasade — COMMENTED** [review]
This seems reasonable given what I understand of the system

**Comment by antgonza** [comment]
reviewed by @wasade, merging.

**Pull request merged**